### PR TITLE
Adds Energy Cake & Cakehat Tweaks

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Added the energy cake, which can be crafted via combining an energy sword and birthday cake. 
I would have added in the functionality to make it a energy cakehat, but I'm not sure if that'd be accepted under the antag freeze. If a mantainer permits it I'll do it. Meanwhile, it's still in the code, just not obtainable.
Cakehat inhands added, Ian can now wear a cakehat (but not an energy cakehat)

## Why It's Good For The Game

More food for the chef is always good, and It'd be pretty funny for a chef to loot an esword from someone just to bake it into a cake. The new cakehat, if it gets approved to be craftable, falls under the same idea; It'd be funny to see used, while being literally worse than the energy dagger, which costs 2tc, being normal-sized and incredibly obvious.

image

## Changelog
:cl:
add: New food item; The Energy Cake! Can be tablecrafted by combining a birthday cake with an energy sword. Careful, it's a bit sharp.
add: Energy cakehats implemented! Microwave an energy cake to create one. The microwaving weakens the embedded energy sword, though.
add: Ian can now wear a cakehat.
imageadd: Added cakehat inhands.
imageadd: Birthday cakes are now animated.
/:cl:

### CODE REVIEW

**Comment by shellspeed1** [comment]
I feel like there should be a toysword version. Than again eh.

**Comment by Tharcoonvagh** [comment]
Fake Energy Cake

File: code/modules/clothing/head/misc_special.dm [hunk 1]
```diff
@@ -60,24 +69,49 @@
...
-	hitsound = 'sound/weapons/tap.ogg'
+	hitsound = hitsound_off
 	STOP_PROCESSING(SSobj, src)
 
 /obj/item/clothing/head/hardhat/cakehat/is_hot()
 	return on * heat
+
+/obj/item/clothing/head/hardhat/cakehat/energycake
+	name = "energy cake"
+	desc = "You put the energy sword on your cake. Brilliant."
```
Comment (PKPenguin321) [inline code/modules/clothing/head/misc_special.dm]:
```suggestion
	desc = "You put an energy sword on your cake. Brilliant."
```

**Comment by PKPenguin321** [comment]
Actually biting into this should put a hole through the roof of your mouth

**Comment by Qustinnus** [comment]
Why doesn't this cake give liquid electricity? It's literaly an energy cake

**Comment by carlarctg** [comment]
good idea, i'll do that when i get home

File: code/modules/clothing/head/misc_special.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/clothing/head/misc_special.dm]: no

**Comment by SkeletalElite** [comment]
> good idea, i'll do that when i get home
> 
> and @PKPenguin321 i don't know how to make food deal damage when eaten (other than like, put crevice spike in it or something)

arnold pizza makes your arm fly off, maybe you can use a similar method to that

**Comment by carlarctg** [comment]
taking a bite of an energy cake will deal 30 damage, and taking a bite of an energy cake slice 18

also of course the other cake pr had to happen at the same time as this

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -682,7 +682,7 @@
 	description = "The blood of Ethereals, and the stuff that keeps them going. Great for them, horrid for anyone else."
 	nutriment_factor = 5 * REAGENTS_METABOLISM
 	color = "#97ee63"
-	taste_description = "pure electrictiy"
+	taste_description = "pure electrictity"
```
Comment (Fikou) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: wouldnt this be electricity

Comment (carlarctg) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: fuc k

Comment (Qustinnus) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: Electric titty

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: feature

Comment (carlarctg) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: no

File: code/modules/food_and_drinks/food/snacks_cake.dm [hunk 3]
```diff
@@ -188,6 +188,48 @@
...
+	name = "energy cake slice"
+	desc = "For the traitor on the go."
+	icon_state = "energycakeslice"
+	force = 2
+	hitsound = 'sound/weapons/blade1.ogg'
+	filling_color = "#00FF00"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 4, /datum/reagent/consumable/sprinkles = 2, /datum/reagent/consumable/nutriment/vitamin = 1,  /datum/reagent/consumable/pwr_game = 2, /datum/reagent/consumable/liquidelectricity = 2)
+	tastes = list("cake" = 3, "a Vlad's Salad" = 1)
+
+/obj/item/reagent_containers/food/snacks/cakeslice/birthday/energy/proc/energy_bite(mob/living/user)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/food_and_drinks/food/snacks_cake.dm]: why is this a proc

Comment (ExcessiveUseOfCobblestone) [reply code/modules/food_and_drinks/food/snacks_cake.dm]: nvm

File: code/modules/food_and_drinks/food/snacks_cake.dm [hunk 4]
```diff
@@ -188,6 +188,48 @@
...
+	to_chat(user, "<font color='red' size='5'>As you eat the cake slice, you accidentally hurt yourself on the embedded energy dagger!</font>")
+	user.apply_damage(18,BRUTE,BODY_ZONE_HEAD)
+	playsound(user, 'sound/weapons/blade1.ogg', 5, TRUE)
+
+/obj/item/reagent_containers/food/snacks/cakeslice/birthday/energy/attack(mob/living/M, mob/living/user)
+	. = ..()
+	energy_bite(M, user)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/food_and_drinks/food/snacks_cake.dm]: is this a dupe of 210-12?

**Pull request merged**